src/pkg/packager/composer/list.go

```diff
@@ -72,14 +103,14 @@ func (ic *ImportChain) append(c types.ZarfComponent, relativeToHead string, vars
 }
 
 // NewImportChain creates a new import chain from a component
-func NewImportChain(head types.ZarfComponent, arch, flavor string) (*ImportChain, error) {
+func NewImportChain(head types.ZarfComponent, index int, arch, flavor string) (*ImportChain, error) {
```

Comment:
Why are we adding an index to track here? My understanding for part of why a double linked list was chosen for composition was so that we don't have to worry about keeping track of an index. 

cc: @Noxsios @Racer159

Reply:
Could we use the component name in the output instead of an index? Might be easier to understand

Reply:
@lucasrod16 This `index` is not the index in the import chain, but rather the component's index in it's source `zarf.yaml`

Reply:
Ah I see, thanks @Noxsios!

Reply:
@lucasrod16 The issue with having the name instead of the index is that two components might have the same name. We see this in our k3s package where we have an arm / amd resulting in two "k3s" components. This also allows the path to be evaluated by yq. Though I could see it being confusing to users when we get into composability. We should do some user testing on this at some point to see how intuitive it is to understand.